website/blog/2020-01-10-cobalt.md

```diff
@@ -0,0 +1,953 @@
+with large codebases that Pants is usually used for.
+
+### Improvements for Bloop integration
+
+In order to make the integration between Bloop and Metals a smoother experience,
+several new improvements are included in this release. There're based on
+feedback from users as well us our own experiences while working with Metals and
+should make the overall experience of importing the build and compiling much
+better.
+
+#### SemanticDB is resolved inside Bloop
+
+Metals requires the SemanticDB compiler plugin for a number of features to
+function properly. Because of that it was quite difficult to add it
+automatically in all supported build tools even if those supported Bloop itself.
+Perfect examples are Seed and Fury, which use Bloop to compile code, however
+adding a plugin automatically would require changes to their core code.
+
+Currently, thanks to
+[recent changes](https://github.com/scalacenter/bloop/pull/942), Metals can send
+information about the SemanticDB version, which Bloop uses to automatically
+download and configure the SemanticDB scalac plugin. Thanks to that no changes
+in the build tool itself are needed. Particular changes in regards to build
+tools are:
+
+- Gradle - predef script code used in automatic import is now simplified a lot
+  and works with even more customized workspaces. The only thing it needs to do
+  is add the Bloop plugin and run `bloopInstall`, which can easily also be done
+  manually by the user.
+- Maven - no manual configuration is now needed. To import automatically just
+  the plugin is needed to be invoked with a setting for downloading sources.
+- Sbt - the `metals-sbt` plugin is now removed and the Bloop plugin is now added
+  via automatically generated `./project/metals.sbt` file and ran when imported
+  automatically. Thanks to this solution it should be possible to reuse Metals
+  compilation in the upcoming feature for offloading compilation in Sbt.
+- Mill - logic for adding semanticDB plugin is removed from Mill.
+- Pants, Fury, Seed - no changes were needed for the semanticDB integration and
+  should work out of the box.
+
+#### Switch to to using Bloop Launcher
+
+Instead of using internal Metals logic, we now use the Bloop Launcher, which
+enables us to either connect to an existing Bsp server or start a new Bloop
+instance with the specified version if none is running. This greatly simplified
+the connection to Bloop and enabled us to work on further improvements.
+
+#### Restart Bloop automatically if newer version is needed
+
+Whenever Metals is started and connects to the Bloop build server it receives
+information about the version it is running. We compare that version to the one
+needed by Metals to make sure everything is properly working. If the current
+running version is older we offer to automatically restart it with the newer
+version. However, this may not work if Bloop was installed manually.
+
+#### Automatically reconnect to Bsp/Bloop if connection is broken
```

Comment:
Are we sure we need level 4 nested headers? I think levels 2 and 3 are usually enough.

Reply:
Reduced to level 3 - I was sure about it easier. I moved the intro just below the `Build tools` header and was able to reduce to 3.